sql/src/test/java/io/crate/execution/engine/collect/files/FileReadingIteratorTest.java

```diff
@@ -91,10 +127,96 @@ public void testIteratorContract() throws Exception {
+
+        List<Object[]> expectedResult = Arrays.asList(
+            new Object[]{new BytesRef(firstLine)},
+            new Object[]{new BytesRef(secondLine)});
+        BatchIteratorTester tester = new BatchIteratorTester(batchIteratorSupplier);
+        tester.verifyResultAndEdgeCaseBehaviour(expectedResult);
+    }
+
+    @Test
+    public void testIteratorContract_givenNoInputFormat_AndNoRelevantFileExtension_thenParsesAsJson() throws Exception {
+        givenTempFileOfFormat(JSON);
+        givenFileUri();
+        Supplier<BatchIterator<Row>> batchIteratorSupplier = () -> createBatchIterator(
+            Collections.singletonList(fileUri), null, null
+        );
+
+        byte[] firstLine = "{\"name\": \"Arthur\", \"id\": 4, \"details\": {\"age\": 38}}".getBytes(StandardCharsets.UTF_8);
+        byte[] secondLine = "{\"id\": 5, \"name\": \"Trillian\", \"details\": {\"age\": 33}}".getBytes(StandardCharsets.UTF_8);
+
+        List<Object[]> expectedResult = Arrays.asList(
+            new Object[]{new BytesRef(firstLine)},
+            new Object[]{new BytesRef(secondLine)});
+        BatchIteratorTester tester = new BatchIteratorTester(batchIteratorSupplier);
+        tester.verifyResultAndEdgeCaseBehaviour(expectedResult);
+    }
+
+    private void givenTempFileOfFormat(InputFormat format) throws IOException {
+        tempFilePath = createTempFile();
+        tmpFile = tempFilePath.toFile();
+
+        switch (format) {
+            case JSON:
+                writeJsonToOutputStream(tmpFile);
+                break;
+            case CSV:
+                writeCsvToOutputStream(tmpFile);
+        }
+    }
+
+    private void givenTempFileWithSuffix(String suffix) throws IOException {
+        tempFilePath = createTempFile("tempfile", suffix);
+        File tmpFile = tempFilePath.toFile();
+        switch (suffix) {
+            case ".json":
+                writeJsonToOutputStream(tmpFile);
+                break;
+            case ".csv":
+                writeCsvToOutputStream(tmpFile);
+                break;
+            default:
+                writeJsonToOutputStream(tmpFile);
+        }
+    }
+
+    private void writeJsonToOutputStream(File tmpFile) throws IOException {
+        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tmpFile), StandardCharsets.UTF_8)) {
+            writer.write("{\"name\": \"Arthur\", \"id\": 4, \"details\": {\"age\": 38}}\n");
+            writer.write("{\"id\": 5, \"name\": \"Trillian\", \"details\": {\"age\": 33}}\n");
+        }
+    }
+
+    private void writeCsvToOutputStream(File tmpFile) throws IOException {
+        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tmpFile), StandardCharsets.UTF_8)) {
+            writer.write("name,id\n");
+            writer.write("Arthur,4\n");
+            writer.write("Trillian,5\n");
+        }
+    }
+
+    private void givenFileUri() {
+        fileUri = tempFilePath.toUri().toString();
```

Comment:
Is there a reason this is a class variable and not local? Kind of unexpected to see it change here, hidden in the method.

Reply:
This was done for readability - so that the given/when/then steps (or arrange/act/assert - whatever you prefer to call them) are clearer to read.